packages/core/src/external/athenahealth/index.ts

```diff
@@ -0,0 +1,331 @@
+import axios, { AxiosInstance, AxiosResponse, AxiosError } from "axios";
+import {
+  Medication,
+  MedicationAdministration,
+  MedicationDispense,
+  MedicationStatement,
+} from "@medplum/fhirtypes";
+import {
+  patientResourceSchema,
+  PatientResource,
+} from "@metriport/shared/interface/external/athenahealth/patient";
+import { errorToString } from "@metriport/shared";
+import { S3Utils } from "../aws/s3";
+import { out } from "../../util/log";
+import { capture } from "../../util/notifications";
+import { Config } from "../../util/config";
+import { uuidv7 } from "../../util/uuid-v7";
+import { createHivePartitionFilePath } from "../../domain/filename";
+
+interface ApiConfig {
+  threeLeggedAuthToken: string;
+  practiceId: string;
+  environment: "api" | "api.preview";
+  clientKey: string;
+  clientSecret: string;
+}
+
+const region = Config.getAWSRegion();
+const responsesBucket = Config.getEhrResponsesBucketName();
+
+function getS3UtilsInstance(): S3Utils {
+  return new S3Utils(region);
+}
+
+export type AthenaMedication = { medication: string; medicationid: number };
+
+export type MedicationWithRefs = {
+  medication: Medication;
+  administration?: MedicationAdministration;
+  dispense?: MedicationDispense;
+  statement?: MedicationStatement;
+};
+
+class AthenaHealthApi {
+  private axiosInstanceFhirApi: AxiosInstance;
+  private axiosInstanceProprietary: AxiosInstance;
+  private twoLeggedAuthToken: string;
+  private threeLeggedAuthToken: string;
+  private practiceId: string;
+  private s3Utils: S3Utils;
+
+  private constructor(private config: ApiConfig) {
+    this.twoLeggedAuthToken = "";
+    this.threeLeggedAuthToken = config.threeLeggedAuthToken;
+    this.practiceId = config.practiceId.replace("a-1.Practice-", "");
```

Comment:
put `a-1.Practice-` in an appropriately named const - have no idea what this means, or why it's being replaces